core/services/pipeline/common_test.go

```diff
@@ -155,3 +156,134 @@ func TestUnmarshalJSONSerializable_Invalid(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckInputs(t *testing.T) {
+	for _, tt := range []struct {
+		name   string
+		inputs []pipeline.Result
+		expErr bool
+	}{
+		{
+			name:   "nil",
+			inputs: nil,
+		},
+		{
+			name:   "empty",
+			inputs: []pipeline.Result{},
+		},
+		{
+			name:   "empty-val",
+			inputs: []pipeline.Result{{}},
+		},
+		{
+			name:   "single",
+			inputs: []pipeline.Result{{Value: "testval"}},
+		},
+		{
+			name:   "multi",
+			inputs: []pipeline.Result{{Value: "testval"}, {Value: float64(1.1)}},
+		},
+		{
+			name:   "single-err",
+			inputs: []pipeline.Result{{Error: errors.New("test error")}},
+			expErr: true,
+		},
+		{
+			name:   "multi-errs",
+			inputs: []pipeline.Result{{Error: errors.New("test error")}, {Error: errors.New("test error")}},
+			expErr: true,
+		},
+		{
+			name:   "multi-val-and-errs",
+			inputs: []pipeline.Result{{Value: "testval"}, {Error: errors.New("test error")}},
+			expErr: true,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			err := pipeline.CheckInputs(tt.inputs)
+			if err != nil && !tt.expErr {
+				t.Errorf("unexpected error: %v", err)
+			} else if err == nil && tt.expErr {
+				t.Error("error expected, but got none")
+			}
+		})
+	}
+}
+
+func TestCheckInputsLen(t *testing.T) {
+	for _, tt := range []struct {
+		name     string
+		inputs   []pipeline.Result
+		min, max int
+		expErr   bool
+	}{
+		{
+			name:   "nil-optional",
+			inputs: nil,
+			min:    0, max: 1,
+		},
+		{
+			name:   "nil-required",
+			inputs: nil,
+			min:    1, max: 1,
+			expErr: true,
+		},
+		{
+			name:   "empty-optional",
+			inputs: []pipeline.Result{},
+			min:    0, max: 1,
+		},
+		{
+			name:   "empty-required",
+			inputs: []pipeline.Result{},
+			min:    1, max: 1,
+			expErr: true,
+		},
+		{
+			name:   "single-optional",
+			inputs: []pipeline.Result{{Value: "testval"}},
+			min:    0, max: 1,
+		},
+		{
+			name:   "single-required",
+			inputs: []pipeline.Result{{Value: "testval"}},
+			min:    1, max: 1,
+		},
+		{
+			name:   "multi-optional",
+			inputs: []pipeline.Result{{Value: "testval"}, {Value: float64(1.1)}},
+			min:    0, max: 2,
+		},
+		{
+			name:   "multi-required",
+			inputs: []pipeline.Result{{Value: "testval"}, {Value: float64(1.1)}},
+			min:    2, max: 2,
+		},
+		{
+			name:   "multi-optional-short",
+			inputs: []pipeline.Result{{Value: "testval"}},
+			min:    0, max: 2,
+		},
+		{
+			name:   "multi-required-short",
+			inputs: []pipeline.Result{{Value: "testval"}},
+			min:    2, max: 2,
+			expErr: true,
+		},
+		{
+			name:   "single-long",
+			inputs: []pipeline.Result{{Value: "testval"}, {Value: float64(1.1)}},
+			min:    1, max: 1,
+			expErr: true,
+		},
```

Comment:
Since CheckInputsLen also ends up calling CheckInputs, also add 1 test case where there is an error in one of the inputs.
And in that case, ensure that the error returned is ErrTooManyErrors. This sort of validates that CheckInputsLen does indeed also check the inputs for errors.